### PR TITLE
connectors-ci: use new domain for test reports

### LINKED ISSUE #27497: connectors-ci: better connector test debugging experience
## Current status
On PR and on nightly build we batch the connector test run: a single workflow logs captures tests output from multiple connector. As tests are executed in parallel it can be hard to follow the test execution of a specific connector. Especially for nightly builds where >80 connectors are tested ([example](https://github.com/airbytehq/airbyte/actions/runs/5317126142)).

## Suggestion
1. Create per connector HTML report pages like **what's demoed [here](https://www.loom.com/share/b6e68012e1ad4127bdc66a25c211d848)**
2. Host these reports in the `airbyte-ci-reports` GCS bucket. With a retention of 45 days (like GHA logs)?
E.G: `gs://airbyte-ci-reports/airbyte-ci/connectors/test/nightly_builds/master/1686655046/25ef3a2686617fe0549871d7453b6da70a546db6/source-file/0.3.11/output.html`
3. Make these URL public
4. Add the URL to the html report to the `output.json` file
5. In bot comments on PR we share links to these report page and not to the GHA workflow logs
6. In the daily #connector-health channel reports we share links to connector specific test reports.

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #27497

Using the `https://storage.cloud.google.com` domain we can implement cookie based authentication to restrict access to the test reprots to a specific set of google email addresses / group.

This PR has no direct effect on the bucket permission, but a 👍 from reviewer will make me:
* Remove public access to the `airbyte-ci-reports-multi` bucket
* Allow airbyters and GL team to be Storage Object viewers

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-openweather test report (commit [`c8fd5ae11b`](https://github.com/airbytehq/airbyte/pull/27826/commits/c8fd5ae11b07073d61a4c717a0a76b1fc5113467)) - ✅

⏲️  Total pipeline duration: 01mn35s seconds

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-openweather/metadata.yaml | ✅        |
| Connector version semver check                                            | ✅        |
| QA checks                                                                 | ✅        |
| Code format checks                                                        | ✅        |
| Connector package install                                                 | ✅        |
| Build source-openweather docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                | ✅        |
| Acceptance tests                                                          | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_private-access-to-reports/1688032794/c8fd5ae11b07073d61a4c717a0a76b1fc5113467/source-openweather/0.1.6/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-openweather test
```

**Review by evantahler — APPROVED** [review]
